### PR TITLE
Create a start page

### PR DESCRIPTION
**Pull request by ZhimingZhao**
Changes:

- Added a home page
- For now start game button leads straight to the problem page

homepage

### CODE REVIEW

File: src/frontend/src/app/startpage/startpage.component.scss [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+.MuiBox-root {
+  align-items: center;
+}
+
+.roundbutton {
+  border-radius: 20px;
+  background: transparent;
+  padding: 6px 12px;
```
Comment (AntonKanug) [inline src/frontend/src/app/startpage/startpage.component.scss:31]:
```suggestion
  padding: 12px;
```

To center text in button

File: src/frontend/src/app/startpage/startpage.component.scss [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  background-color: white;
+  color: black;
+  transform: scale(2);
+}
+
+.MuiBox-root {
+  align-items: center;
+}
+
+.roundbutton {
```
Comment (AntonKanug) [inline src/frontend/src/app/startpage/startpage.component.scss:28]:
```suggestion
.round-button {
```

nit: use '-' for spaces in classes

File: src/frontend/src/app/startpage/startpage.component.html [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+<div class="buttoncontainer">
+  <div
+    style="
+      position: absolute;
+      opacity: 1;
+      transform: translateX(0%) translateZ(0px);
+    "
+  >
+    <div class="roundbutton" [routerLink]="['/problem']">Start Game</div>
```
Comment (AntonKanug) [inline src/frontend/src/app/startpage/startpage.component.html:22]: Try to use ```button``` with ```mat-button```

File: src/frontend/src/app/startpage/startpage.component.html [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      matTooltipPosition="after"
+    >
+      Host room
+    </div>
+    <div
+      class="loginbutton"
+      matTooltip="Coming soon!"
+      matTooltipPosition="after"
+    >
+      <div class="loginicon">
```
Comment (AntonKanug) [inline src/frontend/src/app/startpage/startpage.component.html:35]: icon and right arrow seems misplaced in the image, should fix before demo

File: src/frontend/src/app/startpage/startpage.component.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-startpage',
```
Comment (AntonKanug) [inline src/frontend/src/app/startpage/startpage.component.ts:4]: for future follow, when gen component follow 'word-word' patten

**Review by AntonKanug — CHANGES_REQUESTED** [review]

File: src/frontend/src/app/startpage/startpage.component.scss [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  color: white;
+  background-color: white;
+}
+
+.buttoncontainer {
+  position: relative;
+  display: -webkit-box;
+  display: -webkit-flex;
+  display: -ms-flexbox;
+  display: flex;
```
Comment (Tamas-Leung) [inline src/frontend/src/app/startpage/startpage.component.scss:101]: Don't have to worry about the webkit, ms stuff for flex box. Flexbox widely supported on all browsers now

**Review by Tamas-Leung — APPROVED** [review]

File: src/frontend/src/app/startpage/startpage.component.scss (hunk 6 above)
Comment (ZhimingZhao) [reply src/frontend/src/app/startpage/startpage.component.scss:98]: Removed webkit and ms flexbox

File: src/frontend/src/app/startpage/startpage.component.html (hunk 4 above)
Comment (ZhimingZhao) [reply src/frontend/src/app/startpage/startpage.component.html:35]: image

**Review by AntonKanug — APPROVED** [review]

**Pull request merged**